checker/src/callbacks.rs

```diff
@@ -250,6 +254,23 @@ impl MiraiCallbacks {
         defs: &[DefId],
         analysis_info: &mut AnalysisInfo<'analysis, 'tcx>,
     ) {
+        // Determine the functions we want to analyze.
+        if let Some(fname) = &analysis_info.options.single_func {
+            analysis_info.fn_whitelist = Some(vec![fname.clone()]);
+        } else if analysis_info.options.test_only {
+            // Extract test functions from the main test runner.
+            if let Some((entry_def_id, _)) = tcx.entry_fn(LOCAL_CRATE) {
+                let fns = Self::extract_test_fns(tcx, entry_def_id);
+                if fns.is_empty() {
+                    warn!("Could not extract any tests from main entry point");
+                }
+                analysis_info.fn_whitelist = Some(fns);
+            } else {
+                warn!("Did not find any suitable main entry point to identify tests");
```

Comment:
This raises the useless question of what makes an entry point suitable.

"any suitable" -> "a"